extensions/eclipse-che-theia-plugin-ext/src/browser/che-github-main.ts

```diff
@@ -0,0 +1,105 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { CheGithubMain } from '../common/che-protocol';
+import { interfaces } from 'inversify';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import axios, { AxiosInstance } from 'axios';
+
+export class CheGithubMainImpl implements CheGithubMain {
+    private readonly envVariableServer: EnvVariablesServer;
+    private axiosInstance: AxiosInstance = axios;
+    private apiUrl: string;
+    private token: string | undefined;
+
+    constructor(container: interfaces.Container) {
+        this.envVariableServer = container.get(EnvVariablesServer);
+        this.envVariableServer.getValue('CHE_API').then(variable => {
+            if (variable && variable.value) {
+                this.apiUrl = variable.value;
+                this.getToken().then(token => {
```

Comment:
hello, it looks odd to me to lazily initialize token variable in a constructor.

Reply:
Removed token initialization from constructor